src/clp_ffi_py/PyObjectUtils.hpp

```diff
@@ -47,6 +49,32 @@ using PyObjectPtr = std::unique_ptr<PyObjectType, PyObjectDeleter<PyObjectType>>
  */
 template <typename PyObjectType>
 using PyObjectStaticPtr = std::unique_ptr<PyObjectType, PyObjectTrivialDeleter<PyObjectType>>;
+
+/**
+ * A guard class for Python exceptions. In certain CPython methods, such as `tp_finalize`,
+ * the exception state must remain unchanged throughout execution. This class saves the current
+ * exception state upon initialization and restores it upon destruction, ensuring the exception
+ * status is preserved.
+ * Docs: https://docs.python.org/3/c-api/typeobj.html#c.PyTypeObject.tp_finalize
+ */
+class PyErrGuard {
+public:
+    // Constructor
+    PyErrGuard() = default;
```

Comment:
hmmmm, I think the point is we can rely on the docstring to help readers understand what this class is used for, instead of trying to read the implementation and figure out what's going on.
That said, I don't think we should try to polish the implementation side. Instead, do u think we should improve any documentation of this class?

Reply:
I guess it's ok, it doesn't take very long for me to realize what it is doing.

Reply:
kk, then I will say probably just leave it as it is